Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -94,10 +94,34 @@ actor ExternalBuildSystemAdapter {
   private var lastRestart: Date?
 
   static package func projectRoot(for workspaceFolder: AbsolutePath, options: SourceKitLSPOptions) -> AbsolutePath? {
-    guard localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json")) else {
-      return nil
+    var buildServerConfigLocations = [String]()
+
+    #if os(Windows)
+    let workspacePath = workspaceFolder.pathString + "\\.bsp"
+    let localAppDataPath = ProcessInfo.processInfo.environment["LOCALAPPDATA"] ?? "C:\\Users\\Default\\AppData\\Local\\bsp"
+    let programDataPath = ProcessInfo.processInfo.environment["PROGRAMDATA"] ?? "C:\\ProgramData\\bsp"
+
+    buildServerConfigLocations.append(contentsOf: [workspacePath, localAppDataPath, programDataPath])
+    #elseif os(Linux)
```

Comment:
The spec says `Unix + Mac`, so I think we should search for `XDG_DATA_HOME` also on macOS and not have an `os(Linux)` clause.

Reply:
Yep.